src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java

```diff
@@ -172,26 +174,34 @@ public void start(final Map<String, String> props) {
         return sourceRecords.isEmpty() ? null : sourceRecords;
       } else {
         BsonDocument changeStreamDocument = next.get();
-
         Map<String, String> sourceOffset = new HashMap<>();
         sourceOffset.put("_id", changeStreamDocument.getDocument("_id").toJson());
         if (isCopying.get()) {
           sourceOffset.put("copy", "true");
         }
-
+        JsonWriterSettings jsonOptions = null;
+        try {
+          jsonOptions = handleJsonOptions();
+        } catch (CloneNotSupportedException e) {
+          e.printStackTrace();
+        }
         String topicName =
             getTopicNameFromNamespace(
                 prefix, changeStreamDocument.getDocument("ns", new BsonDocument()));
 
-        Optional<String> jsonDocument = Optional.empty();
+        Optional<BsonDocument> jsonDocumentraw = Optional.empty();
```

Comment:
The name of this variable is misleading as it doesn't hold a JSON document. I suggest renaming it to `bsonDocument`, `originalDocument`, or simply `document`.

Reply:
indeed i'm doing so